src/openassetio-core/src/errors/exceptionMessages.hpp

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 The Foundry Visionmongers Ltd
+#pragma once
+
+#include <optional>
+#include <string>
+
+#include <openassetio/EntityReference.hpp>
+#include <openassetio/access.hpp>
+#include <openassetio/errors/BatchElementError.hpp>
+#include <openassetio/typedefs.hpp>
+
+namespace openassetio {
+inline namespace OPENASSETIO_CORE_ABI_VERSION {
+namespace errors {
+/**Get an error code name as a printable string*/
+Str errorCodeName(BatchElementError::ErrorCode code);
+
+/**Construct a full message to place into a convenience exception.*/
+std::string createBatchElementExceptionMessage(const BatchElementError &err, size_t index,
```

Comment:
[fix] Sorry if I missed them, but looks like this is missing test coverage?

Reply:
The message is tested in `test_manager.py` per API function.  Doesn't look like this function is tested in isolation, but many branches in this function are (currently) unused.  I've added a comment re. possibly removing the `optional`ity.

Reply:
Sorry I'd missed [this bit](https://github.com/OpenAssetIO/OpenAssetIO/pull/1109/files#diff-b6b4a676c1e9833a98eec7842db81decc905f6d5562d8b4bc2785b692a431c6eR2794)